### PR TITLE
Add closed loop BZ#2149030 for Provisioning Templates

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding closed loop [BZ#2149030](https://bugzilla.redhat.com/show_bug.cgi?id=2149030) for Provisioning Templates

Dependent Nailgun PR: https://github.com/SatelliteQE/nailgun/pull/898

### CODE REVIEW

File: pytest_fixtures/component/provisioning_template.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 # Provisioning Template Fixtures
 import pytest
+from box import Box
 from nailgun import entities
+from packaging.version import Version
 
+from robottelo import constants
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+from robottelo.api.utils import wait_for_tasks
```
Comment (ogajduse) [inline pytest_fixtures/component/provisioning_template.py:9]: Do not import these two, use the Host object instead.
See https://github.com/SatelliteQE/robottelo/blob/0e441a0998c7a3208edeb7a843bdda9af4995ee5/tests/foreman/api/test_remoteexecution.py#L64

**Review by ogajduse — APPROVED** [review]
ACK code-wise

File: pytest_fixtures/component/provisioning_template.py (hunk 1 above)
Comment (shubhamsg199) [reply pytest_fixtures/component/provisioning_template.py]: Thanks, Updated.

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioningtemplate.py -k test_positive_template_check_ipxe
nailgun: 898

**Comment by ogajduse** [comment]
@shubhamsg199 do not forget to cherry-pick SatelliteQE/nailgun#898 into z-stream branches. All auto cherry-picks failed.

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioningtemplate.py -k test_positive_template_check_ipxe

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioningtemplate.py -k test_positive_template_check_ipxe

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioningtemplate.py -k test_positive_template_check_ipxe

File: tests/foreman/api/test_provisioningtemplate.py [hunk 2]
```diff
@@ -310,3 +310,53 @@ def test_positive_provision_template_check_net_interface(
...
+    ):
+        """Read the iPXE template and verify 'ks=' parameter is rendered as
+           expected for different rhel hosts.
+
+        :id: 065ef48f-bec5-4535-8be7-d8527fa21563
+
+        :expectedresults: The rendered iPXE template contains the "ks=" parameter
+                          expected for respective rhel hosts.
+
+        :CaseImportance: High
```
Comment (adarshdubey-star) [inline tests/foreman/api/test_provisioningtemplate.py:334]: Non blocking already defined at the module level
```suggestion
        
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**